moveit_ros/perception/handeye_calibration_solver/src/handeye_solver_default.cpp

```diff
@@ -0,0 +1,376 @@
+  python_name = PyString_FromString("handeye.calibrator");
+  python_module = PyImport_Import(python_name);
+  Py_DECREF(python_name);
+  if (!python_module)
+  {
+    ROS_ERROR_STREAM_NAMED(LOGNAME, "Failed to load python module: "
+                                        << "handeye.calibrator");
+    PyErr_Print();
+    return false;
+  }
+
+  // Find handeye.calibrator.HandEyeCalibrator class
+  python_class = PyObject_GetAttrString(python_module, "HandEyeCalibrator");
+  Py_DECREF(python_module);
+  if (!python_class || !PyCallable_Check(python_class))
+  {
+    ROS_ERROR_STREAM_NAMED(LOGNAME, "Can't find \"HandEyeCalibrator\" python class");
+    PyErr_Print();
+    return false;
+  }
+
+  // Parse sensor mount type
+  python_value = PyString_FromString("");
+  if (setup == EYE_TO_HAND)
+    python_value = PyString_FromString("Fixed");
+  else if (setup == EYE_IN_HAND)
+    python_value = PyString_FromString("Moving");
+  if (!python_value)
+  {
+    ROS_ERROR_STREAM_NAMED(LOGNAME, "Can't creat sensor mount type python value");
+    Py_DECREF(python_class);
+    PyErr_Print();
+    return false;
+  }
+
+  // Create handeye.calibrator.HandEyeCalibrator instance
+  python_args = PyTuple_New(1);
+  PyTuple_SetItem(python_args, 0, python_value);
+  if (!python_args)
+  {
+    ROS_ERROR_STREAM_NAMED(LOGNAME, "Can't build python arguments");
+    Py_DECREF(python_class);
+    PyErr_Print();
+    return false;
+  }
+  python_instance = PyEval_CallObject(python_class, python_args);
+  Py_DECREF(python_args);
+  Py_DECREF(python_class);
+  if (!python_instance)
+  {
+    ROS_ERROR_STREAM_NAMED(LOGNAME, "Can't create \"HandEyeCalibrator\" python instance");
+    PyErr_Print();
+    return false;
+  }
+
+  // Find handeye.calibrator.HandEyeCalibrator.add_sample method
+  python_func_add_sample = PyObject_GetAttrString(python_instance, "add_sample");
+  if (!python_func_add_sample || !PyCallable_Check(python_func_add_sample))
+  {
+    ROS_ERROR_STREAM_NAMED(LOGNAME, "Can't find 'add_sample' method");
+    Py_DECREF(python_instance);
+    PyErr_Print();
+    return false;
+  }
+
+  // Add sample poses to handeye.calibrator.HandEyeCalibrator instance
+  size_t number_of_poses = effector_wrt_world.size();
+  PyArrayObject *numpy_arg_eef_wrt_base[number_of_poses], *numpy_arg_obj_wrt_sensor[number_of_poses];
+  PyObject *python_array_eef_wrt_base[number_of_poses], *python_array_obj_wrt_sensor[number_of_poses];
+  PyObject* python_args_sample[number_of_poses];
+  npy_intp dims[2]{ ARRAY_SIZE, ARRAY_SIZE };
+  const int number_of_dims{ 2 };
+  // Using C array to store the pyarray data, which will be automatically freed
+  double c_arr_eef_wrt_world[number_of_poses][ARRAY_SIZE][ARRAY_SIZE];
+  double c_arr_obj_wrt_sensor[number_of_poses][ARRAY_SIZE][ARRAY_SIZE];
+  for (size_t i = 0; i < number_of_poses; ++i)
+  {
+    // Convert effector_wrt_world[i] from Eigen::isometry3d to PyArrayObject
+    if (toCArray(effector_wrt_world[i], c_arr_eef_wrt_world[i]))
```

Comment:
Since this function can fail, and is able to report it, would it be easier to just check for failures instead of success and return on failure? Something like this:
```
if (!toCArray(effector_wrt_world[i], c_arr_eef_wrt_world[i]))
{
  return false:
}
```

Reply:
Done as suggested.